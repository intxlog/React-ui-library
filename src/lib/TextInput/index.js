import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

//import css
import styles from './styles.module.scss'

const TextInput = ({
  id,
  className,
  inlineStyles,
  placeholder,
  disabled,
  defaultValue,
  type,
  onChange,
  onBlur,
  error
}) => {
  return (
    <input    
      id={id}
      style={inlineStyles}
      className={classNames({
        [className]: className,
        [styles.element]: true,
        [styles.error]: error,
        [styles.disabled]: disabled
      })}
      placeholder={placeholder}
      disabled={disabled}
      defaultValue={defaultValue}
      type={type}
      onChange={onChange}
      onBlur={onBlur}
    ></input>
  )
}

//rules for props being passed in
TextInput.propTypes = {
  id: PropTypes.string,
  inlineStyles: PropTypes.object,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  className: PropTypes.string,
  defaultValue: PropTypes.string,
  type: PropTypes.oneOf([`text`, `password`]),
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  error: PropTypes.bool
}

//maintain the name for documentation purposes
TextInput.displayName = 'TextInput'

export default TextInput