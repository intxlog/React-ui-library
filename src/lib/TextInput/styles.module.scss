@import './../../styles/helpers.scss';

.element {
  @extend %element;
  font-size: $fontSize;
  line-height: $lineHeight;
  padding: 5px 10px;
}