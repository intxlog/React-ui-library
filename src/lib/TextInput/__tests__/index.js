import React from 'react'
import { shallow } from 'enzyme'
import TextInput from '..'

describe('TextInput', () => {
  let props = {}
  
  //create a shallow wrapper that can be reused
  let wrapper
  //initialize the components
  beforeEach(() => {
    wrapper = shallow(<TextInput {...props}/>)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})