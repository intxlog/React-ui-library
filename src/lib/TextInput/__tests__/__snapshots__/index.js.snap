// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<input
  className="element"
/>
`;
